test/univariate/continuous/johnsonsu.jl

```diff
@@ -0,0 +1,127 @@
+@testset "JohnsonSU" begin
+
+    d1 = JohnsonSU(0.0, 10.0, -2.0, 3.0)
+    @test d1 isa JohnsonSU{Float64}
+    @test params(d1) == (0.0, 10.0, -2.0, 3.0)
+    @test shape(d1) == 0.0
+    @test scale(d1) == 10.0
+    @test partype(d1) === Float64
+    @test eltype(d1) === Float64
+    @test rand(d1) isa Float64
+
+    @test median(d1) == quantile(d1, 0.5)
+    x = quantile.(d1, [0.25, 0.45, 0.60, 0.80, 0.90])
+    @test all(cdf.(d1, x) .≈ [0.25, 0.45, 0.60, 0.80, 0.90])
+    y = cquantile.(d1, [0.25, 0.45, 0.60, 0.80, 0.90])
+    @test all(ccdf.(d1, y) .≈ [0.25, 0.45, 0.60, 0.80, 0.90])
+
+    @test mean(d1) ≈ 7.581281
+    @test var(d1) ≈ 19.1969485
+    @test skewness(d1) ≈ 0.7780949
+    @test kurtosis(d1) ≈ 0.447512937
+
+    d1 = JohnsonSU(10.0f0, 10.0f0, 1.0f0, 3.0f0)
+    @test d1 isa JohnsonSU{Float32}
+    @test params(d1) == (10.0f0, 10.0f0, 1.0f0, 3.0f0)
+    @test shape(d1) == 10.0f0
+    @test scale(d1) == 10.0f0
+    @test partype(d1) === Float32
+    @test eltype(d1) === Float64
+    @test rand(d1) isa Float64
+
+    d1 = JohnsonSU(1.0, 1, 0, 1)
+    d1 = Base.convert(JohnsonSU{Float64}, JohnsonSU())
+
+    d1 = JohnsonSU()
+    @test d1 isa JohnsonSU{Int}
+    @test params(d1) == (0, 1, 0, 1)
+
+    @test pdf(d1, -Inf) == 0.0
+    @test pdf(d1, Inf) == 0.0
+    @test isnan(pdf(d1, NaN))
+
+    @test logpdf(d1, -Inf) == -Inf
+    @test logpdf(d1, Inf) == -Inf
+    @test isnan(logpdf(d1, NaN))
+
+    @test cdf(d1, -Inf) == 0.0
+    @test cdf(d1, Inf) == 1.0
+    @test isnan(cdf(d1, NaN))
+
+    @test logcdf(d1, -Inf) == -Inf
+    @test logcdf(d1, Inf) == 0.0
+    @test isnan(logcdf(d1, NaN))
+
+    @test ccdf(d1, -Inf) == 1.0
+    @test ccdf(d1, Inf) == 0.0
+    @test isnan(ccdf(d1, NaN))
+
+    @test logccdf(d1, -Inf) == 0.0
+    @test logccdf(d1, Inf) == -Inf
+    @test isnan(logccdf(d1, NaN))
+
+    @test invlogcdf(d1, -Inf) == -Inf
+    @test isnan(invlogcdf(d1, Inf))
+    @test isnan(invlogcdf(d1, NaN))
+
+    @test invlogccdf(d1, -Inf) == Inf
+    @test isnan(invlogccdf(d1, Inf))
+    @test isnan(invlogccdf(d1, NaN))
+
+    @inferred pdf(d1, -Inf32)
```

Comment:
Note that `@inferred` doesn't actually test anything, it just errors if the return type doesn't match the inferred return type. Typically it's used in conjunction with a test of behavior, e.g.
```julia
@test @inferred(pdf(d1, -Inf32)) == 0
```

Reply:
Ah, I didn't know it is commonly used with a behavior test. Nice to know!

However, given that the numerical values are already tested by `test/ref/continuous_test.ref.json`, I suppose it's okay to leave it as is?